src/vs/workbench/contrib/notebook/browser/diff/diffComponents.ts

```diff
@@ -228,6 +228,599 @@ interface IDiffElementLayoutState {
+			modified: modifiedRef.object.textEditorModel,
+		}));
+
+		// Reduces flicker (compute this before setting the model)
+		// Else when the model is set, the height of the editor will be x, after diff is computed, then height will be y.
+		// & that results in flicker.
+		await vm.waitForDiff();
+		this._editor!.setModel(vm);
+
+		const handleViewStateChange = () => {
+			this._editorViewStateChanged = true;
+		};
+
+		const handleScrollChange = (e: editorCommon.IScrollEvent) => {
+			if (e.scrollTopChanged || e.scrollLeftChanged) {
+				this._editorViewStateChanged = true;
+			}
+		};
+
+		this.updateEditorOptionsForWhitespace();
+		this._register(this._editor!.getOriginalEditor().onDidChangeCursorSelection(handleViewStateChange));
+		this._register(this._editor!.getOriginalEditor().onDidScrollChange(handleScrollChange));
+		this._register(this._editor!.getModifiedEditor().onDidChangeCursorSelection(handleViewStateChange));
+		this._register(this._editor!.getModifiedEditor().onDidScrollChange(handleScrollChange));
+
+		const editorViewState = this.viewModel.getSourceEditorViewState() as editorCommon.IDiffEditorViewState | null;
+		if (editorViewState) {
+			this._editor!.restoreViewState(editorViewState);
+		}
+
+		const contentHeight = this._editor!.getContentHeight();
+		this.viewModel.editorHeight = contentHeight;
+	}
+	private updateEditorOptionsForWhitespace() {
+		const editor = this._editor;
+		if (!editor) {
+			return;
+		}
+		const uri = editor.getModel()?.modified.uri || editor.getModel()?.original.uri;
+		if (!uri) {
+			return;
+		}
+		const ignoreTrimWhitespace = this.textConfigurationService.getValue<boolean>(uri, 'diffEditor.ignoreTrimWhitespace');
+		editor.updateOptions({ ignoreTrimWhitespace });
+
+		this._register(this.textConfigurationService.onDidChangeConfiguration(e => {
+			if (e.affectsConfiguration(uri, 'diffEditor') &&
+				e.affectedKeys.has('diffEditor.ignoreTrimWhitespace')) {
+				const ignoreTrimWhitespace = this.textConfigurationService.getValue<boolean>(uri, 'diffEditor.ignoreTrimWhitespace');
+				editor.updateOptions({ ignoreTrimWhitespace });
+			}
+		}));
+	}
+	layout(state: IDiffElementLayoutState) {
+		DOM.scheduleAtNextAnimationFrame(DOM.getWindow(this._diffEditorContainer), () => {
+			if (state.editorHeight && this._editor) {
+				this._editorContainer.style.height = `${this.viewModel.layoutInfo.editorHeight}px`;
+				this._editor.layout({
+					width: this._editor!.getViewWidth(),
+					height: this.viewModel.layoutInfo.editorHeight
+				});
+			}
+
+			if (state.outerWidth && this._editor) {
+				this._editorContainer.style.height = `${this.viewModel.layoutInfo.editorHeight}px`;
+				this._editor.layout();
+			}
+
+			this.layoutNotebookCell();
+		});
+	}
+
+	override dispose() {
+		// The editor isn't disposed yet, it can be re-used.
+		// However the model can be disposed before the editor & that causes issues.
+		if (this._editor) {
+			this._editor.setModel(null);
```

Comment:
the editor handles the dispose already `listenersToRemove.push(model.onWillDispose(() => this.setModel(null)));` so if there is a problem with disposing the model, it might be something else.